### PR TITLE
Add Support for Jobs API

### PR DESCRIPTION
**Pull request by nithinyell**
Ready for review.

### CODE REVIEW

**Comment by kristinpeterson** [comment]
@lucholaf when you have a moment, can you please review?

**Comment by nithinyell** [comment]
@lucholaf  Please review.

File: lib/M2XJob.m [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+static NSString * const kPath = @"/jobs";
+
+@implementation M2XJob
+
+- (NSString *)path {
+    return [NSString stringWithFormat:@"%@/%@", kPath, [self[@"id"] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
+}
+
+-(void)listOfJobs:(M2XArrayCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XJob.m]: this has to be a class method

File: lib/M2XJob.m [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        for (NSDictionary *dict in response.json[@"jobs"]) {
+            M2XJob *job = [[M2XJob alloc] initWithClient:self.client attributes:dict];
+            [array addObject:job];
+        }
+        
+        completionHandler(array, response);
+    }];
+}
+
+-(void)viewJobs:(NSString*)jobID CompletionHandler:(M2XArrayCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XJob.m]: this should be `viewJob` instead of `viewJobs`

File: lib/M2XJob.m [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    [self.client getWithPath:[NSString stringWithFormat:@"%@/%@", kPath,jobID] parameters:nil completionHandler:^(M2XResponse *response) {
+        NSMutableArray *array = [NSMutableArray array];
+        
+        for (NSDictionary *dict in response.json[@"jobs"]) {
```
Comment (lucholaf) [inline lib/M2XJob.m]: why is it parsing as if it were an array? the endpoint returns only the object of the job

File: lib/M2XJob.m [hunk 4]
```diff
@@ -16,32 +16,26 @@ - (NSString *)path {
...
             [array addObject:job];
         }
         
         completionHandler(array, response);
     }];
+    
 }
 
--(void)viewJobs:(NSString*)jobID CompletionHandler:(M2XArrayCallback)completionHandler{
+-(void)viewJob:(NSString*)jobID CompletionHandler:(M2XBaseCallback)completionHandler{
```
Comment (lucholaf) [inline lib/M2XJob.m]: Move this method to `M2XClient` and follow implementation and similar signature of methods like `- (void)deviceWithId:(NSString *)identifier completionHandler:(M2XDeviceCallback)completionHandler`

**Comment by nithinyell** [comment]
Moved method as directed.

**Pull request merged**